doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -78,145 +78,107 @@ Displaying
 
-   This command sets :n:`@option` to :n:`@value`. The original value of ` option` is
-   restored when the current module ends.
 
-   .. TODO : option and value are not syntax entries
+.. cmd:: {? Local | Global | Export } Set @option {| @num | @string }
+   :name: Set @option
 
-   .. cmdv:: Local Set @option @value
+   Sets :token:`option` to the specified value.  Scoping qualifiers are
+   described :ref:`here <set_unset_scope_qualifiers>`.
 
-      This command sets :n:`@option` to :n:`@value`. The
-      original value of :n:`@option` is restored at the end of the module.
+.. cmd:: {? Local | Global | Export } Unset @option
+   :name: Unset @option
 
-   .. cmdv:: Global Set @option @value
+   Sets :token:`option` to its default value.  Scoping qualifiers are
+   described :ref:`here <set_unset_scope_qualifiers>`.
 
-      This command sets :n:`@option` to :n:`@value`. The
-      original value of :n:`@option` is *not* restored at the end of the module.
-      Additionally, if set in a file, :n:`@option` is set to value when the file
-      is `Require`-d.
-
-   .. cmdv:: Export Set @option
-
-      This command set :n:`@option` to :n:`@value`. The original state
-      of :n:`@option` is restored at the end of the current module, but :n:`@option`
-      is set to :n:`@value` when this module is imported.
-
-
-.. cmd:: Unset @option
-
-   This command turns off :n:`@option`.
-
-   .. cmdv:: Local Unset @option
+.. cmd:: Test @option
 
-      This command turns off :n:`@option`. The original state of :n:`@option`
-      is restored when the current *section* ends.
+   Prints the current value of :token:`option`.
 
-   .. cmdv:: Global Unset @option
+.. cmd:: Print Options
 
-      This command turns off :n:`@option`. The original state of :n:`@option`
-      is *not* restored at the end of the module. Additionally, if unset in a file,
-      :n:`@option` is reset to its default value when the file is `Require`-d.
+   Prints the current value of all flags and options, and the names of all tables.
 
-   .. cmdv:: Export Unset @option
 
-      This command turns off :n:`@option`. The original state of :n:`@option`
-      is restored at the end of the current module, but :n:`@option` is set to
-      its default value when this module is imported.
+.. cmd:: Add @table @string
+   :name: Add `table` `value`
 
+   Adds :token:`string` to :token:`table`.
 
-.. cmd:: Test @option
+.. cmd:: Remove @table @string
+   :name: Remove @table @string
 
-   This command prints the current value of :n:`@option`.
+   Removes :token:`string` from :token:`table`.
 
+.. comment THIS DIDN'T WORK, e.g. for "Test Search Blacklist "foo"."
+   cmd:: Test @table @string
+   :name: Test @table @string
 
-.. TODO : table is not a syntax entry
+   Reports whether :token:`table` contains :token:`string`.
 
-.. cmd:: Add @table @value
-   :name: Add `table` `value`
+.. cmd:: Test @table for @string
+   :name: Test @table for @string
```

Comment:
Two kinds of table are currently supported: for string (e.g. "Search Blacklist") and for global reference (e.g. "Printing Coercion"). So, `Test @table for @qualid` should be added.

Reply:
Will do.  Presumably `Add table` and `Remove table` also take qualids.

I wish there was a way to extract this information systematically rather than having to read code and/or rely on reviewers.

Reply:
> Presumably Add table and Remove table also take qualids.

Yes, `Remove ...` also takes a qualid for those table which takes a qualid (like the `Printing ...` tables).

>  wish there was a way to extract this information systematically rather than having to read code and/or rely on reviewers.

You mean that the information is given at the time of the parsing rule? Maybe something is doable in this direction.